controllers/managedocs_controller.go

```diff
@@ -136,3 +145,34 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateComponents(req ctrl.Request) ([]v1.ManagedOCSComponent, error) {
+	var components []v1.ManagedOCSComponent
+
+	// Checking the StorageCluster component.
+	component := v1.ManagedOCSComponent{
+		Name: "StorageCluster",
+	}
+
+	var scList ocsv1.StorageClusterList
+
+	if err := r.List(r.ctx, &scList); err != nil {
+		r.Log.Error(err, "error listing StorageCluster")
+		return nil, err
+	}
```

Comment:
I believe `Get` is better in this case than `list`

Reply:
No a Get would fail if the storage cluster isn't present.

Reply:
Yes, it will fail with a proper return value. 
But if you think it is better to use list instead, I will not insist

Reply:
Where could I find information on the errors that Get returns? If there is a NotFound error type that it returns than I agree that Get would be cleaner, but I can find no documentation on the error these calls make.

Reply:
You need to do `if !errors.IsNotFound(err)` on the returned `err` of the `get` operation to differentiate. 
You can look at the implementation of `CreateOrUpdate` as an example.